x/staking/types/delegation.go

```diff
@@ -130,8 +130,27 @@ func NewUnbondingDelegation(
 
 // AddEntry - append entry to the unbonding delegation
 func (ubd *UnbondingDelegation) AddEntry(creationHeight int64, minTime time.Time, balance math.Int) {
-	entry := NewUnbondingDelegationEntry(creationHeight, minTime, balance)
-	ubd.Entries = append(ubd.Entries, entry)
+	// Check the entries exists with creation_height and complete_time
+	entryIndex := -1
+	for index, ubdEntry := range ubd.Entries {
+		if ubdEntry.CreationHeight == creationHeight && ubdEntry.CompletionTime.Equal(minTime) {
+			entryIndex = index
+			break
+		}
+	}
+	// entryIndex exists
+	if entryIndex != -1 {
+		ubdEntry := ubd.Entries[entryIndex]
+		ubdEntry.Balance = ubdEntry.Balance.Add(balance)
+		ubdEntry.InitialBalance = ubdEntry.InitialBalance.Add(balance)
```

Comment:
@gsk967 Can you remind me when do Balance and InitialBalance differ?

Reply:
I believe it's when the validator gets slashed (I think?)

Reply:
@amaurym  When validator get slashed  all the unbonding delegations `balance` will decreased by `slashing_factor * initial_balance `

https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/slash.go#L176-L222